lib/bolt/project_migrator/modules.rb

```diff
@@ -0,0 +1,222 @@
+          puppetfile.resolve
+        rescue Bolt::Error => e
+          @outputter.print_migrate_step("#{e.message} Skipping module migration.")
+          return false
+        end
+
+        migrate_managed_modules(puppetfile, puppetfile_path, moduledir)
+
+        # Move remaining modules to 'modules'
+        consolidate_modules(modulepath)
+
+        # Delete old modules that are now managed
+        delete_modules(modulepath.first, puppetfile.modules)
+
+        # Add modules to project
+        update_project_config(modules.map(&:to_hash), config)
+      end
+
+      # Migrates the managed modules. If modules were selected to be managed,
+      # the Puppetfile is rewritten and modules are installed. If no modules
+      # were selected, the Puppetfile is deleted.
+      #
+      private def migrate_managed_modules(puppetfile, puppetfile_path, moduledir)
+        if puppetfile.modules.any?
+          # Show the new Puppetfile content
+          message  = "Generated new Puppetfile content:\n\n"
+          message += puppetfile.modules.map(&:to_spec).join("\n").to_s
+          @outputter.print_migrate_step(message)
+
+          # Write Puppetfile
+          @outputter.print_migrate_step("Updating Puppetfile at #{puppetfile_path}")
+          puppetfile.write(puppetfile_path)
+
+          # Install Puppetfile
+          @outputter.print_migrate_step("Syncing modules from #{puppetfile_path} to #{moduledir}")
+          Bolt::Puppetfile::Installer.new({}).install(puppetfile_path, moduledir)
+        else
+          @outputter.print_migrate_step(
+            "Project does not include any managed modules, deleting Puppetfile "\
+            "at #{puppetfile_path}"
+          )
+          FileUtils.rm(puppetfile_path)
+        end
+      end
+
+      # Prompts the user to select modules, returning a list of
+      # the selected modules.
+      #
+      private def select_modules(modules)
+        @outputter.print_migrate_step(
+          "Select modules that are direct dependencies of your project. Bolt will "\
+          "automatically manage dependencies for each module selected, so do not "\
+          "select a module's dependencies unless you use content from it directly "\
+          "in your project."
+        )
+
+        all = prompt("Select all modules? ([y]es/[n]o) ")
+        return modules if all
+
+        modules.select do |mod|
+          prompt("Select #{mod.title}? ([y]es/[n]o) ")
+        end
+      end
+
+      # Prompts the user for a choice until they choose a valid
+      # option. Returns the response mapped to the choice.
+      #
+      private def prompt(prompt)
+        choices = {
+          'y'   => true,
+          'yes' => true,
+          'n'   => false,
+          'no'  => false
+        }
+
+        loop do
+          @outputter.print_prompt(prompt)
+          response = $stdin.gets.to_s.downcase.chomp
+          return choices[response] if choices.key?(response)
```

Comment:
This is a small thing but it might be nice to have an else here if we get anything other than 'y | yes | n | no':
```
    Select puppetlabs-ruby_task_helper? ([y]es/[n]o) y
    Select puppetlabs-ruby_plugin_helper? ([y]es/[n]o) y
    Select puppetlabs-stdlib? ([y]es/[n]o) f
    Select puppetlabs-stdlib? ([y]es/[n]o) f
    Select puppetlabs-stdlib? ([y]es/[n]o) 
    Select puppetlabs-stdlib? ([y]es/[n]o) f
    Select puppetlabs-stdlib? ([y]es/[n]o) f
    Select puppetlabs-stdlib? ([y]es/[n]o) f
    Select puppetlabs-stdlib? ([y]es/[n]o) f
    Select puppetlabs-stdlib? ([y]es/[n]o) f
    Select puppetlabs-stdlib? ([y]es/[n]o) f
    Select puppetlabs-stdlib? ([y]es/[n]o) yno
    Select puppetlabs-stdlib? ([y]es/[n]o) y
    Select puppetlabs-aws_inventory? ([y]es/[n]o) y
```

Just something in red that's `'Please use 'y' or 'n''` kind of thing